src/interpolations.jl

```diff
@@ -915,6 +920,37 @@ function value(ip::Lagrange{3,RefPrism,2}, i::Int, ξ::Vec{3})
     throw(ArgumentError("no shape function $i for interpolation $ip"))
 end
 
+
+#####################################
+# Lagrange dim 3 RefPyramid order 1 #
+#####################################
+# Build on https://defelement.com/elements/examples/pyramid-Lagrange-1.html
+
+getnbasefunctions(::Lagrange{3,RefPyramid,1}) = 5
+
+vertexdof_indices(::Lagrange{3,RefPyramid,1}) = ((1,), (2,), (3,), (4,), (5,),)
+facedof_indices(::Lagrange{3,RefPyramid,1}) = ((1,4,3,2), (1,2,5), (2,3,5), (3,4,5), (4,1,5), )
+edgedof_indices(::Lagrange{3,RefPyramid,1}) = ((1,2), (2,3), (3,4), (4,1), (1,5), (2,6), (3,5), (4,6))
+
+function reference_coordinates(::Lagrange{3,RefPyramid,1})
+    return [Vec{3, Float64}((0.0, 0.0, 0.0)),
+            Vec{3, Float64}((1.0, 0.0, 0.0)),
+            Vec{3, Float64}((1.0, 1.0, 0.0)),
+            Vec{3, Float64}((0.0, 1.0, .0)),
+            Vec{3, Float64}((0.0, 0.0, 1.0))]
+end
+
+function value(ip::Lagrange{3,RefPyramid,1}, i::Int, ξ::Vec{3})
+    (x,y,z) = ξ
+    zzero = z ≈ 1.0
+    i == 1 && return zzero ? 0.0 : (-x*y+(z-1)*(-x-y-z+1))/(z-1)
```

Comment:
This means the derivative can't be evaluated at `z = 1.0` (truncating the dual part of the input, so it will have 0 derivative).

Reply:
At least not until ForwardDiff 0.16.

But this is discontinuous around z = 1? Is that intended?

Reply:
I used this as ref https://defelement.com/elements/examples/pyramid-Lagrange-1.html
I think that the pyramid is degenerate at the fifth node. But note that N_I = 1 at node I, and zero at the other nodes is still true for pyramids

Maybe @termi-official has some insight regarding pyramid elements

Reply:
One of the basis functions is $-xy/(z-1)$. So yes, you have a singularity at $z=1$ (note $x=y=0$). IIRC this is a problem with the way how these pyramidal elements are constructed. If you want the derivative at this specific point you can formulate y and x as a function of z and apply l'Hospital.

Reply:
Wieners, C. (1997). Conforming discretizations on tetrahedrons, pyramids, prisms and hexahedrons. Preprint, University of Stuttgart. has a proof that there exists no continuous differentiable (bi-)linear function on pyramids.